### ISSUE TITLE
[$250] LHN - Group chat room without message stays on LHN

### ISSUE THREAD

**Issue by vincdargento** [Expensify/App#53496]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.70-6
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** testpayment935+737@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Navigate to https://staging.new.expensify.com
2. Create a group chat with 1 user
3. Notice the group created was shown in LHN
4. Navigate to other chat
5. Notice the group is no more shown in LHN
6. Change the group name or Send a message to a group and delete it 
7. Notice the group still shown in LHN

## Expected Result:
As the group chat have no chat history or system message it should not be shown in LHN

## Actual Result:
The group conversation shown in LHN without any message or history

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/5729b6db-dda4-4e3f-b9c9-b9407165e0ea

https://github.com/user-attachments/assets/644c24d0-eb03-4f37-b343-2964eaac9813

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021864491578413816640</li>
        <li>Upwork Job ID: 1864491578413816640</li>
        <li>Last Price Increase: 2024-12-19</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @AndrewGable</details>

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The group conversation shown in LHN without any message or history

### What is the root cause of that problem?

After we rename the report, `notificationPreference` of the user is updated to `always`. So it stays in LHN.

<img width="1512" alt="Screenshot 2024-12-04 at 15 00 08" src="https://github.com/user-attachments/assets/a0fae068-507f-4825-8b86-ef80fb4383d9">

### What changes do you think we should make in order to solve the problem?

In `updateGroupChatName` function, we should update `notificationPreference` of the current user to `always` if it's `hidden`.

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?



### What alternative solutions did you explore? (Optional)
To clarify, I think we can add a new action when we edit the group chat name like we do when we update the policy room name

1. We can define a new action type for updating the group chat name and create an optimistic action when we update the group chat name like we do [here](https://github.com/Expensify/App/blob/ce282d38777e9e0a267e1b6f44bff9c4aace98ee/src/libs/actions/Report.ts#L2438)

https://github.com/Expensify/App/blob/ce282d38777e9e0a267e1b6f44bff9c4aace98ee/src/libs/actions/Report.ts#L684

2. Define a new function to get the translated message for the new action. And update a case for this action in LHN, `ReportActionItem`, `getReportName`, `copyToClipBoard`.

**Comment by usamazeeshan5** [comment]
1: What is the root cause of that problem?
left hand navigations not updating properly.
2: What changes do you think we should make in order to solve the problem?
Implement a check in updateGroupChatName to handle this logic. This ensures that 
only the required groups appear in the LHN based on the user's preferences.
3:What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?
Multiple rename operations.
Renaming after modifying other group properties like participants or settings.
Groups with no history or messages.

**Comment by usamazeeshan5** [comment]
1: What is the root cause of that problem?
left hand navigations not updating properly.
2: What changes do you think we should make in order to solve the problem?
Implement a check in updateGroupChatName to handle this logic. This ensures that 
only the required groups appear in the LHN based on the user's preferences.
3:What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?
Multiple rename operations.
Renaming after modifying other group properties like participants or settings.
Groups with no history or messages.

**Comment by sobitneupane** [comment]
Thanks for the proposal @nkdengineer 

Can you please add more details in RCA and solution section of your proposal.

**Comment by sobitneupane** [comment]
Waiting for proposal.

**Comment by sobitneupane** [comment]
Waiting for proposal.

**Comment by sobitneupane** [comment]
> After we rename the report, notificationPreference of the user is updated to always from the backend. So it stays in LHN even we click on another report.

@nkdengineer So, the report appears in the LHN because the backend sets the notificationPreference to "always," correct?

> In updateGroupChatName function, we should update notificationPreference of the current user in optimisticData to always if it's hidden.

How does this solve the issue? Wouldn't we want the notificationPreference to remain "hidden"? IMO, the backend shouldn't change the notificationPreference to "always" when renaming the group chat if the intention is to keep it hidden.

The same should apply to sending and deleting messages in a group. If all messages in a group are deleted, and the group has no messages, its notificationPreference should be set to "hidden."

**Comment by nkdengineer** [comment]
> @nkdengineer So, the report appears in the LHN because the backend sets the notificationPreference to "always," correct?

Yes.

@sobitneupane I think we can ask internal about the expected behavior. If `hidden` is expected we should fix it from the backend side.

If `always` is expected, what do you think about my alternative solution that will add a system message when we update the group chat name

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by nkdengineer** [comment]
@AndrewGable Can you take a look at this [commtent](https://github.com/Expensify/App/issues/53496#issuecomment-2556461012)? Does it make sense if we create a system message action when we update the group chat name?

**Comment by sobitneupane** [comment]
https://github.com/Expensify/App/issues/53496#issuecomment-2556461012

cc: @AndrewGable

**Comment by sobitneupane** [comment]
https://github.com/Expensify/App/issues/53496#issuecomment-2556461012

cc: @AndrewGable

**Comment by AndrewGable** [comment]
Seems like it is internal, I agree

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by adelekennedy** [comment]
Updated issue to internal, also going to try to reproduce here

**Comment by adelekennedy** [comment]
This is still reproducible for me, the empty chat disappears and the edited chat remains in the LHN after it's empty so I assume we ant to keep this open to add the system message in

**Comment by adelekennedy** [comment]
@sobitneupane an you reproduce? I now can't seem to

**Comment by adelekennedy** [comment]
and knocking off the overdue here

**Comment by sobitneupane** [comment]
Yep. I was also unable to reproduce the issue.

**Comment by jponikarchuk** [comment]
QA team still can repro this issue in the latest build v9.1.19-3

https://github.com/user-attachments/assets/7e88d572-153d-40cf-9f0c-45b5eb9589a7

https://github.com/user-attachments/assets/792d1e6e-a0a7-47b2-86ba-e98bbea644bf

**Comment by AndrewGable** [comment]
So @sobitneupane - Is this still an internal issue?

**Comment by sobitneupane** [comment]
@AndrewGable Yup, I believe it should be handled in the backend. When the first message is sent, the notification preference changes from 'hidden' to 'always,' and even if the first message is deleted, it remains 'always.'

**Comment by adelekennedy** [comment]
internal issue - dropping priority

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week)

**Comment by adelekennedy** [comment]
With that message above should we close as this is low priority?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Fourth week)
